Lib/gftools/builder/__init__.py

```diff
@@ -0,0 +1,341 @@
+            self.logger.info("Creating variable fonts from %s" % source)
+            sourcebase = os.path.splitext(os.path.basename(source))[0]
+            args["output_path"] = os.path.join(
+                self.config["vfDir"], sourcebase + "-VF.ttf",
+            )
+            output_files = self.run_fontmake(source, args)
+            newname = self.rename_variable(output_files[0])
+            self.post_process(newname)
+            all_variables[source] = newname
+        if all_variables:
+            self.gen_stat(all_variables)
+
+    def run_fontmake(self, source, args):
+        if "output_dir" in args:
+            original_output_dir = args["output_dir"]
+            tmpdir = tempfile.TemporaryDirectory()
+            args["output_dir"] = tmpdir.name
+
+        if source.endswith(".glyphs"):
+            FontProject().run_from_glyphs(source, **args)
+        elif source.endswith(".designspace"):
+            FontProject().run_from_designspace(source, **args)
+        elif source.endswith(".ufo"):
+            FontProject().run_from_ufos([source], **args)
+        else:
+            raise ValueError("Can't build from unknown source file: %s" % source)
+        if "output_path" in args:
+            return [args["output_path"]]
+        else:
+            # Move it to where it should be...
+            file_names = os.listdir(args["output_dir"])
+            for file_name in file_names:
+                shutil.move(
+                    os.path.join(args["output_dir"], file_name), original_output_dir
+                )
+            tmpdir.cleanup()
+            args["output_dir"] = original_output_dir
+            return [os.path.join(original_output_dir, x) for x in file_names]
+
+    def rename_variable(self, fontfile):
+        if "axisOrder" not in self.config:
+            font = TTFont(fontfile)
+            self.config["axisOrder"] = [ax.axisTag for ax in font["fvar"].axes]
+        axes = ",".join(self.config["axisOrder"])
+        newname = fontfile.replace("-VF.ttf", "[%s].ttf" % axes)
+        os.rename(fontfile, newname)
+        return newname
+
+    def gen_stat(self, varfonts):
+        filenames = varfonts.values()
+        if len(filenames) > 1 and "ital" not in self.config["axisOrder"]:
+            # *Are* any italic? Presumably, but test
+            if any([ "italic" in x.lower() for x in filenames ]):
+                self.config["axisOrder"].append("ital")
+
+        if "stylespaceFile" in self.config and self.config["stylespaceFile"]:
+            self.gen_stat_stylespace(self.config["stylespaceFile"], filenames)
+        elif "stat" in self.config:
+            self.gen_stat_config(self.config["stat"], varfonts)
+        else:
+            ttFonts = [TTFont(x) for x in filenames]
+            gen_stat_tables(ttFonts, self.config["axisOrder"])
+            for filename, ttFont in zip(filenames, ttFonts):
+                ttFont.save(filename)
+
+    def gen_stat_config(self, stat, varfonts):
```

Comment:
This is great. Imo, building a STAT table from a yaml file is so good it warrants its own function in `gftools.stat`. I can then reuse it in `gftools gen-stat` by adding an optional file argument.

I can happily do this at a later point.

Reply:
I think let's do that now.